lsp-mode.el

```diff
@@ -3621,9 +3621,24 @@ If ACTION is not set it will be selected from `lsp-code-actions'."
   (lsp-execute-code-action-by-kind "source.organizeImports"))
 
 (defun lsp--apply-formatting (edits)
-  (let ((lsp--server-sync-method 'full))
-    (save-excursion
-      (lsp--apply-text-edits edits))))
+  (if (fboundp 'replace-buffer-contents)
+      (let ((current-buffer (current-buffer)))
+        (with-temp-buffer
+          (insert-buffer-substring-no-properties current-buffer)
+          (let ((lsp--server-sync-method 'full))
+            (save-excursion
+              (lsp--apply-text-edits edits)))
+          (let ((temp-buffer (current-buffer)))
+            (with-current-buffer current-buffer
+              (replace-buffer-contents temp-buffer)))))
+    (let ((point (point))
+          (w-start (window-start)))
+      (let ((lsp--server-sync-method 'full))
+        (save-excursion
```

Comment:
this save excursion should not be needed since the point is restored manually.

Reply:
I see. Thank you.